src/engine/threaded_engine.h

```diff
@@ -311,6 +312,12 @@ class ThreadedEngine : public Engine {
    */
   void ExecuteOprBlock(RunContext run_ctx, OprBlock *opr_block) {
     ThreadedOpr* threaded_opr = opr_block->opr;
+#if MXNET_USE_CUDA
+    if(run_ctx.ctx.dev_mask() == gpu::kDevMask) {
+      // Signify to kernel that GPU is being used
+      mxnet::op::mxnet_op::KernelState::SetUsingGPU(true);
```

Comment:
this should be put in lazy alloc queue

Reply:
ok

Reply:
The queue doesn't know that it's executing GPU, right?  Are you suggesting to set the queue to know that it should call this function? That seems kind of messy, right?